oak/server/rust/oak_runtime/src/node/http/tests.rs

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use super::*;
+use maplit::hashmap;
+use oak_abi::{label::Label, proto::oak::application::ApplicationConfiguration};
+use std::{collections::HashMap, thread};
+
+#[tokio::test]
+async fn test_low_level_server_node() {
+    let configuration = ApplicationConfiguration {
+        wasm_modules: hashmap! {},
+        initial_node_configuration: None,
+    };
+    info!("Create runtime for test");
+    let runtime = crate::runtime::RuntimeProxy::create_runtime(
+        &configuration,
+        &crate::GrpcConfiguration::default(),
+    );
+
+    let (init_receiver, invocation_receiver) = create_communication_channel(&runtime);
+
+    // Create http server node
+    let server_node = Box::new(
+        HttpServerNode::new(
+            "test-node",
+            HttpServerConfiguration {
+                address: "[::]:2525".to_string(),
+            },
+        )
+        .expect("Could not create server node"),
+    );
+
+    // Start the server node
+    let (notify_sender, notify_receiver) = tokio::sync::oneshot::channel::<()>();
+    let runtime_proxy = runtime.clone();
+    let server_node_thread =
+        std::thread::spawn(move || server_node.run(runtime_proxy, init_receiver, notify_receiver));
```

Comment:
I think this happens because `#[tokio::test]` creates a new tokio runtime and executes the test in that. The server node also then tries to create a tokio runtime for running hyper.

Reply:
I have tried different things and cannot make it work without `#[tokio::test]` or spawning threads. Do you have any suggestions?

Reply:
I don't think it is possible to do it without significant changes. Even if you do it without `#[tokio::test]`, you would still need to spawn a different thread to host the service. As part of a future refactoring to not create a new runtime for the pseudo nodes, it might be possible to run the node via the tokio runtime used in `#[tokio::test]`.

Reply:
I expected replacing the use of the runtime in the HTTP server pseudo-node with a variant of `spawn` from `tokio` would work with `#[tokio::test]`. But I could not make that work. I guess, as you said, that needs more significant changes.